adapters/handlers/rest/authz/handlers_authz.go

```diff
@@ -169,7 +180,24 @@ func (h *authZHandlers) removePermissions(params authz.RemovePermissionsParams,
 		return authz.NewRemovePermissionsBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(fmt.Errorf("invalid permission %s", err.Error())))
 	}
 
-	if err := h.authorizer.Authorize(principal, authorization.UPDATE, authorization.Roles(*params.Body.Name)...); err != nil {
+	role, err := h.controller.GetRoles(*params.Body.Name)
+	if err != nil {
+		return authz.NewRemovePermissionsInternalServerError().WithPayload(cerrors.ErrPayloadFromSingleErr(err))
+	}
+
+	if len(role) != 1 {
+		return authz.NewRemovePermissionsInternalServerError().WithPayload(cerrors.ErrPayloadFromSingleErr(err))
+	}
```

Comment:
**thought (if-minor):** is it needed  🤔 , i think we can just bailout and if there is no role, or we check if the user has role read permission and return 404

Reply:
It's really a "just-in-case" check as you're right that it shouldn't happen. It's mainly if somehow more than one role is returned for the given permission

Reply:
gotcha, last request could oyu update the error to error msg to something say role count was not expected or so , so thing to say the count returned not expected without exposing the role existence or count ?